lib/index.d.ts

```diff
@@ -0,0 +1,528 @@
+     */
+    deviceAuthorization(parameters?: IDeviceAuthParameters, extras?: IDeviceAuthExtras): Promise<IDeviceFlowHandle>
+
+    [key: string]: any
+  }
+
+  export class Client implements IClient {
+    constructor (metadata: IClientMetadata, jwks?: object)
+
+    metadata: IClientMetadata
+    authorizationUrl (parameters?: IAuthorizationUrlParams): string
+    endSessionUrl (parameters?: IEndSessionUrlParams): string
+    callbackParams (input: string | IncomingMessage | Http2ServerRequest): CallbackParamsType
+    callback(redirectUri: string, parameters: CallbackParamsType, checks?: IExtendedCallbackChecks, extras?: ICallbackExtras) : Promise<TokenSet>
+    oauthCallback(redirectUri: string, parameters: CallbackParamsType, checks?: ICallbackChecks, extras?: ICallbackExtras) : Promise<TokenSet>
+    refresh(refreshToken: TokenType, extras?: IRefreshExtras) : Promise<TokenSet>
+    userinfo(accessToken: TokenType) : Promise<object>
+    grant(body: IGrantBody, extras?: IGrantExtras) : Promise<TokenSet>
+    introspect(token: string, tokenTypeHint?: string, extras?: IIntrospectExtras) : Promise<object>
+    revoke(token, tokenTypeHint?: string, extras?: IRevokeExtras) : Promise<void>
+    requestObject(payload: IRequestObjectPayload): Promise<string>
+    deviceAuthorization(parameters?: IDeviceAuthParameters, extras?: IDeviceAuthExtras): Promise<IDeviceFlowHandle>
+    static register(metadata: object, other?: IRegisterOther): Promise<IClient>
+    static fromUri(registrationClientUri: string, registrationAccessToken: string, jwks?: object): Promise<IClient>
+  }
+
+  export interface IDeviceFlowHandleParams {
+    client: IClient
+    exchangeBody: object
+    clientAssertionPayload: object
+    response: string
+    maxAge: number
+  }
+
+  export interface IDeviceFlowHandle {
+    poll() : Promise<TokenSet>
+    expired() : boolean
+    expires_at: number
+    client: IClient
+    maxAge: number
+    exchangeBody: object
+    clientAssertionPayload: object
+    response: string
+    interval: number
+    user_code: string
+    device_code: string
+    verification_uri: string
+    verification_uri_complete: string
+    expires_in: number
+  }
+
+  export class DeviceFlowHandle implements IDeviceFlowHandle {
+    constructor (params : IDeviceFlowHandleParams)
+    poll() : Promise<TokenSet>
+    expired() : boolean
+    // tslint:disable-next-line:variable-name
+    expires_at: number
+    client: IClient
+    maxAge: number
+    exchangeBody: object
+    clientAssertionPayload: object
+    response: string
+    interval: number
+    // tslint:disable-next-line:variable-name
+    user_code: string
+    // tslint:disable-next-line:variable-name
+    device_code: string
+    // tslint:disable-next-line:variable-name
+    verification_uri: string
+    // tslint:disable-next-line:variable-name
+    verification_uri_complete: string
+    // tslint:disable-next-line:variable-name
+    expires_in: number
+  }
```

Comment:
same removals as above, in the `DeviceFlowHandle` i don't see the point duplicating everything from the Interface.